benches/http_core_end_to_end.rs

```diff
@@ -371,12 +362,20 @@ impl Opts {
             req
         };
 
-        let client_ref = &client;
+        let shared_client = &client;
 
         let send_request = async |req| {
-            let res = match client_ref {
-                Client::Http1(tx) => tx.send_request(req).await.expect("client wait h1"),
-                Client::Http2(tx) => tx.send_request(req).await.expect("client wait h2"),
+            let res = match shared_client {
+                Client::Http1(tx) => {
+                    let mut tx = tx.lock().await;
+                    tx.ready().await.expect("client is ready");
+                    tx.send_request(req).await.expect("client wait h1")
+                }
+                Client::Http2(tx) => {
+                    let mut tx = tx.clone();
```

Comment:
Are you sure we want to clone this? I seems to have done this in the past I think and that proxy project using rama seemed to have significant increase of resources by cloning it instead of locking it.

Which is this revert commit: https://github.com/plabayo/rama/commit/e17dc0172b72d444d4135b308551a82dd20e3fa5

Reply:
Ah but wait nvm this is not the core client, this is a bench, where you are cloning a channel. Ok should be okay I guess